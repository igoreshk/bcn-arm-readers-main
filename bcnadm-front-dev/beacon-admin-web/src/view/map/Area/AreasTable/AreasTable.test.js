import React from 'react';
import { shallow } from 'enzyme';
import { AreasTable } from './index';

const props = {
  close: jest.fn(),
  data: [
    {
      description: 'FIRST DESCRIPTION',
      entityId: '5bf2afcc31d7221230a09ea4',
      image: 'blob:http://example.com/1',
      levelId: '5beea5a30bd42e1fb05bc0ea',
      name: 'FIRST NAME'
    },
    {
      description: 'SECOND DESCRIPTION',
      entityId: '5bf2afbc31d7221230a09ea3',
      image: 'blob:http://example.com/2',
      levelId: '5beea5a30bd42e1fb05bc0ea',
      name: 'SECOND NAME'
    },
    {
      description: 'THIRD DESCRIPTION',
      entityId: '5beea89b0bd42e1fb05bc0ef',
      image: 'blob:http://example.com/3',
      levelId: '5beea5a30bd42e1fb05bc0ea',
      name: 'THIRD NAME'
    }
  ],
  editArea: jest.fn(),
  filter: '',
  isAreasLoading: false,
  levelNumber: 1,
  setFilter: jest.fn(),
  toggleSortAreasByField: jest.fn(),
  translate: jest.fn()
};

const wrapper = shallow(<AreasTable {...props} />);

describe('AreasTable', () => {
  it('Should match snapshot', () => expect(wrapper).toMatchSnapshot());
});
