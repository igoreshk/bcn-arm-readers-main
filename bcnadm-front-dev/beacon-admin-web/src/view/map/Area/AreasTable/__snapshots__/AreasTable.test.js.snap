// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreasTable Should match snapshot 1`] = `
<div
  className="areasTable"
>
  <div
    className="header"
  >
    <span
      className="levelNumber"
    >
      <br />
      (
      )
    </span>
    <SearchPanel
      className="areasSearch"
      textFieldRest={
        Object {
          "InputProps": Object {
            "disableUnderline": true,
          },
          "fullWidth": true,
        }
      }
    />
  </div>
  <WithStyles(ForwardRef(Table))
    className="table"
  >
    <WithStyles(ForwardRef(TableHead))>
      <AreasTableHeader
        isAreasLoading={false}
        toggleSortAreasByField={[MockFunction]}
        translate={
          [MockFunction] {
            "calls": Array [
              Array [
                "operator.listOfAreas",
              ],
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
              Array [
                "areas.name",
              ],
              Array [
                "areas.description",
              ],
              Array [
                "areas.name",
              ],
              Array [
                "areas.description",
              ],
              Array [
                "areas.name",
              ],
              Array [
                "areas.description",
              ],
              Array [
                "admin.close",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
    </WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableBody))>
      <WithStyles(ForwardRef(TableRow))
        key="5bf2afcc31d7221230a09ea4"
      >
        <WithStyles(ForwardRef(TableCell))>
          FIRST NAME
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          FIRST DESCRIPTION
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          className="editColumn"
        >
          <Memo(ForwardRef(EditIcon))
            className="editIcon"
          />
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow))
        key="5bf2afbc31d7221230a09ea3"
      >
        <WithStyles(ForwardRef(TableCell))>
          SECOND NAME
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          SECOND DESCRIPTION
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          className="editColumn"
        >
          <Memo(ForwardRef(EditIcon))
            className="editIcon"
          />
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow))
        key="5beea89b0bd42e1fb05bc0ef"
      >
        <WithStyles(ForwardRef(TableCell))>
          THIRD NAME
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          THIRD DESCRIPTION
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          className="editColumn"
        >
          <Memo(ForwardRef(EditIcon))
            className="editIcon"
          />
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
  <div
    className="closeButtonContainer"
    id="buttonRoot"
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[MockFunction]}
    />
  </div>
</div>
`;
